src/landscape.cpp

```diff
@@ -787,7 +787,7 @@ void RunTileLoop()
 		_tile_type_procs[GetTileType(tile)]->tile_loop_proc(tile);
 
 		/* Get the next tile in sequence using a Galois LFSR. */
-		tile = (tile >> 1) ^ (-(int32_t)(tile & 1) & feedback);
+		tile = (static_cast<uint32_t>(tile) >> 1) ^ (-(int32_t)(static_cast<uint32_t>(tile) & 1) & feedback);
```

Comment:
The changes in this file deal with tiles, not dates. Should it be a separate commit or maybe a separate PR?

(Same goes for changes in `newgrf_house.cpp`, `newgrf_industries.cpp`, `newgrf_town.cpp`, `town_cmd.cpp`, `yapf_node.hpp`, and three files in `saveload/*`, just to make a nice list for you of what I found. 😉)

Reply:
No, sadly, these where it runs an operator on the StrongTypeDef, like `&`, `>>`, etc, has to be done here. Before this PR, that was a valid operator. But I think it has very little definition .. what type does `date >> 1` return .. so I revoked access to all those kind of operators.

As a downside, it means that the `TileIndex` also no longer can use them. And those you listed. So they have to be in this PR!

Reply:
Okay! Maybe a separate commit, then? ("No" is a valid answer. 😃)

Reply:
That would be really really difficult to make that a sane and compilable commit :) I rather keep it like this. It is only a few lines :)